### DISCUSSION TITLE
No more load function in error pages

### DISCUSSION THREAD

**Discussion by kevinrenskers** [sveltejs/kit#6157] [Q&A] [answered]
After the load refactor, I now have a problem with my error pages, where I can't access any data from the layout.

Rich said this in https://github.com/sveltejs/kit/discussions/5748:

> Error components — +error.svelte — will no longer be able to use load, as this is a source of bugginess and confusion for the sake of an arguably undesirable feature.

Okay, but how can I now show my error page with the data as I always have in the past?

For example my error template used to look like this:

```
<script context="module">
  export function load({ stuff, error, status }) {
    const updatedStuff = stuff;
    updatedStuff.error = error;
    updatedStuff.status = status;

    return {
      props: updatedStuff,
      stuff: updatedStuff,
    };
  }
</script>

<script>
  import { page } from "$app/stores";

  const campaignId = $page.params.campaignId;

  export let fetchedCampaigns;
  export let status;
  export let error;

  $: campaign = $fetchedCampaigns[campaignId];
</script>

<svelte:head>
  <title>Error 404 | Characters | {campaign.name} | Critical Notes</title>
</svelte:head>

<h1>Error {status}</h1>
<p>{error.message}</p>
```

So the real work happened in the layout file's load function, which would put it into `stuff`, and the error page could grab it from `stuff` and show a nice title. But after the refactor I added `export let data` to `+error.svelte`, and it's `null`. I can't create `+error.js` either.

So I am really wondering how you are supposed to use any data from the layout's load function in the error page, or if that is simply not possible at all anymore.

**Comment by StrawHatHacker** [comment, marked as answer]
https://github.com/sveltejs/kit/issues/6106#issuecomment-1221364121

**Reply by kevinrenskers** [reply]
Thanks, I added a comment there.
